libmattermost.c

```diff
@@ -2036,29 +2064,58 @@ mm_got_hello_user_statuses(MattermostAccount *ma, JsonNode *node, gpointer user_
 	g_list_free(ids);
 }
 
-static void 
+static void
+mm_got_user_statuses_response(MattermostAccount *ma, JsonNode *node, gpointer user_data)
+{
+
+	if (!mm_check_mattermost_response(ma,node,_("Error"),_("Error getting user statuses"),TRUE)) return;
+
+	JsonArray *users = json_node_get_array(node);
+	guint i, len = json_array_get_length(users);
+
+	for (i = 0; i < len; i++) {
+		JsonObject *user = json_array_get_object_element(users,i);
+		const gchar *user_id = json_object_get_string_member(user, "user_id");
+		const gchar *status = json_object_get_string_member(user, "status");
+		const gchar *username = g_hash_table_lookup(ma->ids_to_usernames, user_id);
+
+		if (username != NULL && status != NULL) {
+			purple_protocol_got_user_status(ma->account, username, status, NULL);
+		}
+	}
+}
+
+static void
 mm_refresh_statuses(MattermostAccount *ma, const gchar *id)
 {
-	JsonObject *obj;
-	JsonObject *data;
 	JsonArray *user_ids;
-
-	obj = json_object_new();
-	data = json_object_new();
+	gchar *url;
+	gchar *postdata;
 	user_ids = json_array_new();
 
-	if (id != NULL) {		
+	if (id != NULL) {
 		json_array_add_string_element(user_ids, id);
-		json_object_set_array_member(data, "user_ids", user_ids);
-		json_object_set_string_member(obj, "action", "get_statuses_by_ids");
-		json_object_set_object_member(obj, "data", data);
 	} else {
-		json_object_set_string_member(obj, "action", "get_statuses");
+		GSList *buddy_it;
+		buddy_it = purple_find_buddies(ma->account, NULL);
+		while(buddy_it != NULL){
+			PurpleBuddy *buddy = buddy_it->data;
+			const gchar *buddy_name = purple_buddy_get_name(buddy);
+			const gchar *user_id = g_hash_table_lookup(ma->usernames_to_ids, buddy_name);
+			json_array_add_string_element(user_ids, user_id);
+
+			buddy_it = g_slist_next(buddy_it);
+		}
+		g_slist_free(buddy_it);

```

Comment:
This will leak memory, need to retain the original pointer from the `purple_find_buddies` callback to free?

Reply:
done, updated that.